### ISSUE TITLE
.gitignore for the mypy_cache not created on Windows 10

### ISSUE THREAD

**Issue by charlesbaynham** [python/mypy#8756]
Should mypy create a `.gitignore` file in the `.mypy_cache` folder? This is what other tools like `tox` and `pytest` do. A .gitignore containing

```
*
```

Would save many users the annoyance of having to backtrack their git histories after running mypy and accidentally committing the cache. 

I've seen https://github.com/github/gitignore/pull/2371 but most users don't use this template gitignore and, generally, I think tools should aim to set themselves up, not require the user to add additional configuration elsewhere. 

If you agree, I can make a PR.

**Comment by JukkaL** [comment]
Mypy already does this:

```
$ cat .mypy_cache/.gitignore
# Automatically created by mypy
*
```

If you have an existing `.mypy_cache` directory, you may have to delete it first and allow mypy to recreate it.

**Comment by charlesbaynham** [comment]
Hi @JukkaL , thanks for the reply. In that case, I'd better reopen this as a bug since on my system this isn't happening. 

Python 3.8.2, installed via conda on Windows 10
mypy 0.770, installed via conda

To reproduce:

Setup:
```
mkdir test
cd test
git init
echo "print('hello world')" > test.py
git add .
git commit -m "Start repo"
```
Version info:
```
$ python -c "import sys; print(sys.version)"
mypy --version3.8.2 (default, Apr 14 2020, 19:01:40) [MSC v.1916 64 bit (AMD64)]
$ python -c "import sys; print(sys.platform)"
win32
$ python -c "import sys; print(sys.version_info)"
sys.version_info(major=3, minor=8, micro=2, releaselevel='final', serial=0)
$ mypy --version
mypy 0.770
```
Start structure:
```
$ ls -la
drwxr-xr-x 1 nnn 1049089  0 May  4 11:42 ./
drwxr-xr-x 1 nnn 1049089  0 May  4 11:42 ../
drwxr-xr-x 1 nnn 1049089  0 May  4 11:42 .git/
-rw-r--r-- 1 nnn 1049089 21 May  4 11:42 test.py
```

Run mypy:
```
$ mypy test.py
Success: no issues found in 1 source file

$ ls -la .mypy_cache
total 8
drwxr-xr-x 1 cb6 1049089 0 May  4 11:43 ./
drwxr-xr-x 1 cb6 1049089 0 May  4 11:43 ../
drwxr-xr-x 1 cb6 1049089 0 May  4 11:43 3.8/
(systematics)

$ git status
On branch master
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .mypy_cache/
```

**Comment by ethanhs** [comment]
I can repro this issue on Windows. I'll take a look at what is causing this later.

**Comment by ethanhs** [comment]
Oh this was fixed in #8529.
